### PR TITLE
Rewrite `TestCaseSource` attribute implementation to fix edge cases

### PR DESCRIPTION
**Pull request by frenzibyte**
Breakage noticed in `TestSceneOsuTooltip` on test browser, specifically a test backed by a `TestCaseSource` attribute which points to an array of `string`s. Included similar test in `TestSceneTestWithSource`.

To explain briefly why that breaks with the current code: `string` implements `IEnumerable<char>`, and the logic we have treats types with `IEnumerable` specially and assumes it's an enumerable of multiple parameters per test case, meanwhile in this case the test only intends to accept a single `string` parameter.

Regardless, I've rewritten the entire logic to just rely completely on NUnit and not reinvent the wheel with parsing the attribute.

### CODE REVIEW

File: osu.Framework/Testing/TestBrowser.cs [hunk 1]
```diff
@@ -430,7 +430,7 @@ private void finishLoad(TestScene newTest, Action onCompletion)
                 {
                     string repeatSuffix = i > 0 ? $" ({i + 1})" : string.Empty;
 
-                    var methodWrapper = new MethodWrapper(m.GetType(), m);
+                    var methodWrapper = new MethodWrapper(m.DeclaringType, m);
```
Comment (frenzibyte) [inline osu.Framework/Testing/TestBrowser.cs:433]: Somehow this has been wrong for however many years yet it never became a problem until now. I guess locating source values is its only purpose.

**Review by smoogipoo — APPROVED** [review]
Though I wonder if we shouldn't just be using `new DefaultSuiteBuilder().BuildFrom(new TypeWrapper(newTest.GetType()));` rather than implementing all the reflection ourselves...
